mobius-core/src/test/java/com/spotify/mobius/EffectRouterBuilderImplTest.java

```diff
@@ -0,0 +1,246 @@
+/*
+ * -\-\-
+ * Mobius
+ * --
+ * Copyright (c) 2017-2018 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+package com.spotify.mobius;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import com.spotify.mobius.functions.Consumer;
+import com.spotify.mobius.functions.Function;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+import javax.annotation.Nonnull;
+import org.junit.Before;
+import org.junit.Test;
+
+public class EffectRouterBuilderImplTest {
+
+  private TestConsumer<Event> eventConsumer;
+  private AtomicBoolean ranSimpleEffect;
+
+  private EffectRouterBuilderImpl<Effect, Event> builder;
+  public static final Runnable DUMMY_ACTION =
+      new Runnable() {
+        @Override
+        public void run() {}
+      };
+
+  @Before
+  public void setUp() throws Exception {
+    eventConsumer = new TestConsumer<>();
+    ranSimpleEffect = new AtomicBoolean(false);
+    builder = new EffectRouterBuilderImpl<>();
+  }
+
+  @Test
+  public void integrationTest() throws Exception {
+    final AtomicReference<String> ewpReference = new AtomicReference<>();
+    final Event event = new Event();
+
+    Connection<Effect> connection =
+        Mobius.<Effect, Event>subtypeEffectHandler()
+            .add(EffectWithEvent.class, eventEmitter(event))
+            .add(
+                SimpleEffect.class,
+                new Runnable() {
```

Comment:
Can't we use lambdas instead?

Reply:
In test code yes, but not inanything that ends up in artifacts since it breaks the android java 8 desugaring

Reply:
Done.